src/servers/http_server_v2.cc

```diff
@@ -501,54 +501,88 @@ HTTPAPIServerV2::ResponseRelease(
-    int* counter)
+    std::vector<T>* base, const DataType dtype,
+    const rapidjson::Value& payload_data, int* counter)
 {
   for (size_t i = 0; i < payload_data.Size(); i++) {
     // If last dimension
     if (!payload_data[i].IsArray()) {
       switch (dtype) {
-        case TYPE_BOOL:
+        case TYPE_BOOL: {
+          uint8_t* data_vec = reinterpret_cast<uint8_t*>(base->data());
           data_vec[*counter] = (uint8_t)payload_data[i].GetBool();
           break;
-        case TYPE_UINT8:
+        }
+        case TYPE_UINT8: {
+          uint8_t* data_vec = reinterpret_cast<uint8_t*>(base->data());
           data_vec[*counter] = (uint8_t)payload_data[i].GetInt();
           break;
-        case TYPE_UINT16:
+        }
+        case TYPE_UINT16: {
+          uint16_t* data_vec = reinterpret_cast<uint16_t*>(base->data());
           data_vec[*counter] = (uint16_t)payload_data[i].GetInt();
           break;
-        case TYPE_UINT32:
+        }
+        case TYPE_UINT32: {
+          uint32_t* data_vec = reinterpret_cast<uint32_t*>(base->data());
           data_vec[*counter] = (uint32_t)payload_data[i].GetInt();
           break;
-        case TYPE_UINT64:
+        }
+        case TYPE_UINT64: {
+          uint64_t* data_vec = reinterpret_cast<uint64_t*>(base->data());
           data_vec[*counter] = (uint64_t)payload_data[i].GetInt();
           break;
-        case TYPE_INT8:
+        }
+        case TYPE_INT8: {
+          int8_t* data_vec = reinterpret_cast<int8_t*>(base->data());
           data_vec[*counter] = (int8_t)payload_data[i].GetInt();
           break;
-        case TYPE_INT16:
+        }
+        case TYPE_INT16: {
+          int16_t* data_vec = reinterpret_cast<int16_t*>(base->data());
           data_vec[*counter] = (int16_t)payload_data[i].GetInt();
           break;
-        case TYPE_INT32:
+        }
+        case TYPE_INT32: {
+          int32_t* data_vec = reinterpret_cast<int32_t*>(base->data());
           data_vec[*counter] = (int32_t)payload_data[i].GetInt();
           break;
-        case TYPE_INT64:
+        }
+        case TYPE_INT64: {
+          int64_t* data_vec = reinterpret_cast<int64_t*>(base->data());
           data_vec[*counter] = (int64_t)payload_data[i].GetInt();
           break;
-        case TYPE_FP32:
+        }
+        case TYPE_FP32: {
+          float* data_vec = reinterpret_cast<float*>(base->data());
           data_vec[*counter] = (float)payload_data[i].GetFloat();
           break;
-        case TYPE_FP64:
+        }
+        case TYPE_FP64: {
+          double* data_vec = reinterpret_cast<double*>(base->data());
           data_vec[*counter] = (double)payload_data[i].GetDouble();
           break;
+        }
+        case TYPE_STRING: {
+          const char* cstr = payload_data[i].GetString();
+          uint32_t len = strlen(cstr);
+          int base_len = base->size();
+          base->resize(base_len + sizeof(uint32_t) + len);
```

Comment:
`resize()` is potentially expensive.. i.e. filling new space, and copies due to reallocations. Maybe better to do two pass on `payload_data`, first for getting total byte size and the second for the copies.

Reply:
Done